### PR TITLE
Custom Name Fixes

### PR DESCRIPTION
**Pull request by FoxxoTrystan**
# Description

This is an emercency fix regarding Custom Name, apprently this is fucked... wow...
Regarding DB issues please @DEATHB4DEFEAT review or check if you can assist or anyone.

Thank you.

---

# Changelog

:cl:
- fix: Custom Specie Name!

### CODE REVIEW

File: Content.Shared/Preferences/HumanoidCharacterProfile.cs [hunk 1]
```diff
@@ -534,9 +534,24 @@ public void EnsureValid(ICommonSession session, IDependencyCollection collection
...
+            }
+            else
+            {
+                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename);
+            }
+
+            if (!speciesPrototype.CustomName)
+            {
+                customspeciename = "";
+            }
```
Comment (VMSolidus) [inline Content.Shared/Preferences/HumanoidCharacterProfile.cs:554]:
```suggestion
            if (string.IsNullOrEmpty(Customspeciename))
                customspeciename = "";
            else if (Customspeciename.Length > MaxNameLength)
                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename)[..MaxNameLength];
            else
                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename);

            if (!speciesPrototype.CustomName)
                customspeciename = "";
```

**Review by VMSolidus — CHANGES_REQUESTED** [review]

File: Content.Shared/Preferences/HumanoidCharacterProfile.cs (hunk 1 above)
Comment (FoxxoTrystan) [reply Content.Shared/Preferences/HumanoidCharacterProfile.cs:554]: note: i just reverted what caused the issues, not made it better so this pehaps could be.

File: Content.Shared/Preferences/HumanoidCharacterProfile.cs [hunk 2]
```diff
@@ -534,9 +534,16 @@ public void EnsureValid(ICommonSession session, IDependencyCollection collection
...
+            string customspeciename;
+            if (string.IsNullOrEmpty(Customspeciename))
+                customspeciename = "";
+            else if (Customspeciename.Length > MaxNameLength)
+                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename)[..MaxNameLength];
+            else
+                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename);
+
+            if (!speciesPrototype.CustomName)
+                customspeciename = "";
```
Comment (Mnemotechnician) [inline Content.Shared/Preferences/HumanoidCharacterProfile.cs:546]: For the love of god, just do
```suggestion
            var customspeciename = speciesPrototype.CustomName ? FormattedMessage.RemoveMarkup(Customspeciename ?? "") : "";
                
            if (Customspeciename.Length > MaxNameLength)
                customspeciename = customspeciename[..MaxNameLength];
```

Comment (FoxxoTrystan) [reply Content.Shared/Preferences/HumanoidCharacterProfile.cs:546]: This resulted in return null and crashed.

Comment (DEATHB4DEFEAT) [reply Content.Shared/Preferences/HumanoidCharacterProfile.cs:546]:
```suggestion
            var customspeciename =
                !speciesPrototype.CustomName
                || string.IsNullOrEmpty(Customspeciename)
                    ? ""
                    : Customspeciename.Length > MaxNameLength
                        ? FormattedMessage.RemoveMarkup(Customspeciename)[..MaxNameLength]
                        : FormattedMessage.RemoveMarkup(Customspeciename);
```

**Comment by Mnemotechnician** [comment]
It's done. Do note that even though the checks have passed, we still need someone to actually test the changes (and especially make sure the custom specie name migration works).

**Review by Peptide90 — APPROVED** [review]

**Pull request merged**